retriever/lib/store.js

```diff
@@ -84,62 +85,67 @@ export async function logRetrievalResult(env, params) {
  *
  * @param {Env} env - Cloudflare Worker environment with D1 DB binding
  * @param {string} clientAddress - The address of the client making the request
- * @param {string} rootCid - The root CID to look up
+ * @param {string} pieceCid - The piece CID to look up
  * @returns {Promise<{
- *   ownerAddress: string
- *   pieceRetrievalUrl: string
- *   proofSetId: string
+ *   storageProviderAddress: string
+ *   serviceUrl: string
+ *   dataSetId: string
  * }>}
  */
-export async function getOwnerAndValidateClient(env, clientAddress, rootCid) {
+export async function getStorageProviderAndValidateClient(
+  env,
+  clientAddress,
+  pieceCid,
+) {
   const query = `
-   SELECT ir.set_id, lower(ips.owner) as owner, ipsr.payer, ipsr.with_cdn, pu.piece_retrieval_url, wd.is_sanctioned
-   FROM indexer_roots ir
-   LEFT OUTER JOIN indexer_proof_sets ips
-     ON ir.set_id = ips.set_id
-   LEFT OUTER JOIN indexer_proof_set_rails ipsr
-     ON ir.set_id = ipsr.proof_set_id
-   LEFT OUTER JOIN provider_urls as pu
-     ON lower(ips.owner) = pu.address
-   LEFT OUTER JOIN wallet_details as wd
-     ON lower (ipsr.payer) = wd.address
-   WHERE ir.root_cid = ?
+   SELECT pieces.data_set_id, data_sets.storage_provider_address, data_sets.payer_address, data_sets.with_cdn, providers.service_url, wallet_details.is_sanctioned
+   FROM pieces
+   LEFT OUTER JOIN data_sets
+     ON pieces.data_set_id = data_sets.id
+   LEFT OUTER JOIN providers
+     ON data_sets.storage_provider_address = providers.beneficiary_address
```

Comment:
I think we are missing an index `ON providers(beneficiary_address)` to speed up this query.

We haven't had such an index before, so I guess it's ok to leave it out of this PR.
Ref: https://github.com/filcdn/worker/pull/224#discussion_r2306984553